httpx/_urlparse.py

```diff
@@ -432,13 +432,12 @@ def is_safe(string: str, safe: str = "/") -> bool:
         if char not in NON_ESCAPED_CHARS:
             return False
 
-    # Any '%' characters must be valid '%xx' escape sequences.
-    return string.count("%") == len(PERCENT_ENCODED_REGEX.findall(string))
+    return True
 
 
-def quote(string: str, safe: str = "/") -> str:
+def percent_encoded(string: str, safe: str = "/") -> str:
     """
-    Use percent-encoding to quote a string if required.
+    Use percent-encoding to quote a string.
     """
     if is_safe(string, safe=safe):
         return string
```

Comment:
I suggest use minimal filtering instead of separated one-time function.
```py
    if all(char in NON_ESCAPED_CHARS for char in string):
        return string
```

I also noticed `NON_ESCAPED_CHARS` name in `is_safe` function has different value than it is in `percent_encoded`.

Reply:
Sure. I'd like to treat performance improvements separately to this.
I'll followup with some further work once we've got the behavioural changes in.